src/main/java/spoon/reflect/visitor/filter/AllMethodsSameSignatureFunction.java

```diff
@@ -0,0 +1,208 @@
+ *
+ * The fact that you are presently reading this means that you have had
+ * knowledge of the CeCILL-C license and that you accept its terms.
+ */
+package spoon.reflect.visitor.filter;
+
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import spoon.SpoonException;
+import spoon.reflect.code.CtLambda;
+import spoon.reflect.declaration.CtExecutable;
+import spoon.reflect.declaration.CtMethod;
+import spoon.reflect.declaration.CtType;
+import spoon.reflect.visitor.chain.CtConsumableFunction;
+import spoon.reflect.visitor.chain.CtConsumer;
+import spoon.reflect.visitor.chain.CtQuery;
+import spoon.reflect.visitor.chain.CtQueryAware;
+import spoon.support.visitor.ClassTypingContext;
+import spoon.support.visitor.SubInheritanceHierarchyResolver;
+
+/**
+ * Returns all methods/lambdas with same signature in related inheritance hierarchies.
+ * It can be be used to found all other methods, which has to be changed if signature of method or lambda expression has to be changed.<br>
+ *
+ * Expects {@link CtExecutable} as input
+ * and produces all {@link CtExecutable}s,
+ * which have same signature and are declared in sub/super classes or sub/super interfaces of this or related inheritance hierarchy.<br>
+ *
+ * It makes sense to call this mapping functions for {@link CtMethod} and {@link CtLambda} instances
+ * and then it returns {@link CtMethod} and {@link CtLambda} instance which overrides each other or have same signature.
+ */
+public class AllMethodsSameSignatureFunction implements CtConsumableFunction<CtExecutable<?>>, CtQueryAware {
+
+	private boolean includingSelf = false;
+	private boolean includingLambdas = true;
+	private CtQuery query;
+
+	public AllMethodsSameSignatureFunction() {
+	}
+
+	/**
+	 * @param includingSelf if true then input element is sent to output too. By default it is false.
+	 */
+	public AllMethodsSameSignatureFunction includingSelf(boolean includingSelf) {
+		this.includingSelf = includingSelf;
+		return this;
+	}
+
+	/**
+	 * @param includingLambdas if true then extra search for {@link CtLambda} executables,
+	 * with same signature will be processed too.
+	 * If false, then it returns only {@link CtMethod} instances.
+	 * By default it is true.
+	 */
+	public AllMethodsSameSignatureFunction includingLambdas(boolean includingLambdas) {
+		this.includingLambdas = includingLambdas;
+		return this;
+	}
+
+	@Override
+	public void apply(CtExecutable<?> targetExecutable, final CtConsumer<Object> outputConsumer) {
+		final LambdaFilter lambdaFilter = new LambdaFilter();
+		final List<CtExecutable<?>> targetMethods = new ArrayList<>();
+		targetMethods.add(targetExecutable);
+		if (includingSelf) {
+			outputConsumer.accept(targetExecutable);
+			if (query.isTerminated()) {
+				return;
+			}
+		}
+		CtMethod<?> targetMethod;
+		if (targetExecutable instanceof CtLambda) {
+			targetMethod = ((CtLambda) targetExecutable).getOverriddenMethod();

```

Comment:
not covered (see [coverage](https://coveralls.io/builds/11635765/source?filename=src%2Fmain%2Fjava%2Fspoon%2Freflect%2Fvisitor%2Ffilter%2FAllMethodsSameSignatureFunction.java))

because startMethod is never a lambda in your test?